frontend/src/components/category-stats.tsx

```diff
@@ -0,0 +1,317 @@
+    "var(--mantine-color-violet-6)",
+    "var(--mantine-color-orange-6)",
+    "var(--mantine-color-teal-6)",
+  ];
+
+  if (loading) {
+    return (
+      <Stack gap="md">
+        <Skeleton height={300} />
+        <Skeleton height={300} />
+      </Stack>
+    );
+  }
+
+  if (error) {
+    return (
+      <Alert color="red" title="Error">
+        Failed to load course statistics: {error.message}
+      </Alert>
+    );
+  }
+
+  if (!stats || stats.length === 0) {
+    return (
+      <Alert color="blue" title="No Data Available">
+        No grade statistics are available for this category. This may be because:
+        <ul>
+          <li>The category doesn't have any associated Euclid codes</li>
+          <li>Course data hasn't been loaded for the associated courses</li>
+          <li>Statistics are not available for the courses in this category</li>
+        </ul>
+      </Alert>
+    );
+  }
+
+  return (
+    <Stack gap="lg">
+      <Box>
+        <Title order={3} mb="md">
+          Course Marks Over Time
+        </Title>
+        <Text size="sm" c="dimmed" mb="md">
+          Hover over data points to see mean marks, standard deviation, and course organiser information.
+          Vertical lines mark course organiser changes.
+        </Text>
+        <Paper withBorder p="md">
+          <LineChart
+            h={400}
+            data={chartData}
+            dataKey="year"
+            series={courseCodes.map((code, index) => ({
+              name: code,
+              color: colors[index % colors.length],
+              strokeWidth: 2,
+            }))}
+            referenceLines={referenceLines}
+            curveType="linear"
+            withLegend
+            withTooltip
+            withDots
+            gridAxis="xy"
+            yAxisProps={{
+              domain: [0, 100],
+              tickFormatter: (value) => `${Math.round(value)}%`,
+            }}
+            tooltipProps={{
+              content: ({ active, payload, label }) => {
+                if (active && payload && payload.length) {
+                  return (
+                    <Paper p="sm" withBorder shadow="md">
+                      <Text size="sm" fw={600} mb="xs">{label}</Text>
+                      {payload.map((entry: any, index: number) => {
+                        const courseCode = entry.dataKey;
+                        const meanMark = entry.value;
+                        
+                        // Get the standard deviation and organiser for this course and year
+                        const dataPoint = chartData.find(d => d.year === label);
+                        const stdDev = dataPoint ? dataPoint[`${courseCode}_std`] : null;
+                        const organiser = dataPoint ? dataPoint[`${courseCode}_organiser`] : null;
+                        
+                        return (
+                          <Box key={index}>
+                            <Text size="sm" style={{ color: entry.color }}>
```

Comment:
<img width="282" height="214" alt="Screenshot 2025-09-28_05-27-12" src="https://github.com/user-attachments/assets/14d3b3f6-f19a-4364-a0a6-b7eb207687eb" />

The light mode course code is quite faded, could you increase the contrast a little bit? The course organiser colors are good, and so are all the colors in dark mode.